zkvm/precompiles/src/utils.rs

```diff
@@ -0,0 +1,92 @@
+pub trait CurveOperations {
+    const GENERATOR: [u32; 16];
+    fn add_assign(limbs: &mut [u32; 16], other: &[u32; 16]);
+    fn double(limbs: &mut [u32; 16]);
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct AffinePoint<C: CurveOperations> {
+    pub(crate) limbs: [u32; 16],
+    _marker: std::marker::PhantomData<C>,
+}
+
+impl<C: CurveOperations + Copy> AffinePoint<C> {
+    const GENERATOR: [u32; 16] = C::GENERATOR;
+
+    pub const fn generator_in_affine() -> Self {
+        Self {
+            limbs: Self::GENERATOR,
+            _marker: std::marker::PhantomData,
+        }
+    }
+
+    pub fn new(limbs: [u32; 16]) -> Self {
+        Self {
+            limbs,
+            _marker: std::marker::PhantomData,
+        }
+    }
+
+    // expects the bytes to be in little-endian order.
```

Comment:
nit: for a comment like this, its best to use /// and give a short clean description